spec-dtslint/operators/zipAll-spec.ts

```diff
@@ -0,0 +1,24 @@
+import { of } from 'rxjs';
+import { zipAll } from 'rxjs/operators';
+
+it('should infer correctly', () => {
+  const o = of(of(1, 2, 3)).pipe(zipAll()); // $ExpectType Observable<number[]>
+});
+
+it('should support projecting values', () => {
+  const o = of(of(1, 2, 3)).pipe(zipAll(value => String(value))); // $ExpectType Observable<string>
+});
+
+it('should enforce types', () => {
+  const o = of(1, 2, 3).pipe(zipAll()); // $ExpectError
```

Comment:
Can we add a couple more here for other types of Observable inputs? Iterators, Promises, Arrays, etc.

Reply:
Forgot about this, here you go 😅 
I had to remove the following check

```ts
const p = of(of(1, 2, 3)).pipe(zipAll((value: string) => value)); // $ExpectError
```